spinoso-string/src/lib.rs

```diff
@@ -1762,13 +1496,13 @@ impl String {
     #[inline]
     #[must_use]
     pub fn chop(&mut self) -> bool {
-        if self.buf.is_empty() {
+        if self.is_empty() {
             return false;
         }
-        let bytes_to_remove = if self.buf.ends_with(b"\r\n") {
+        let bytes_to_remove = if self.inner.as_slice().ends_with(b"\r\n") {
```

Comment:
maybe a new `ends_with()` fn on the encoded implementations would make this even nicer

Reply:
👍 Let's address this in a followup.

Reply:
Filed:

- https://github.com/artichoke/artichoke/issues/1721